examples/ex_joystick_events.c

```diff
@@ -115,9 +115,18 @@ static void draw_joystick_button(ALLEGRO_JOYSTICK *joy, int button, bool down)
 
    if (joy) {
       const char *name = al_get_joystick_button_name(joy, button);
-      if (strlen(name) < 4) {
-         al_draw_text(font, fg, x + 13, y + 8, ALLEGRO_ALIGN_CENTRE, name);
+      char name2[4];
+      if (strlen(name) >= 4) {
```

Comment:
is this not the case for a strncpy?

strncpy(name2, name, 3);

Reply:
I don't think so - strncpy will not copy the final 0.